<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>glaxnimate::gui::StrokeStyleWidget</class>
 <widget class="QWidget" name="StrokeStyleWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>262</width>
    <height>284</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <layout class="QVBoxLayout" name="main_layout">
   <item>
    <widget class="QTabWidget" name="tab_widget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Style</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Width</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="spin_stroke_width">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Stroke Width</string>
         </property>
         <property name="maximum">
          <double>999.990000000000009</double>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Cap</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <layout class="QHBoxLayout" name="layout_cap">
         <item>
          <widget class="QToolButton" name="button_cap_butt">
           <property name="toolTip">
            <string>Butt</string>
           </property>
           <property name="text">
            <string>Butt</string>
           </property>
           <property name="icon">
            <iconset theme="stroke-cap-butt">
             <normaloff>.</normaloff>.</iconset>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QToolButton" name="button_cap_round">
           <property name="toolTip">
            <string>Round</string>
           </property>
           <property name="text">
            <string>Round</string>
           </property>
           <property name="icon">
            <iconset theme="stroke-cap-round">
             <normaloff>.</normaloff>.</iconset>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QToolButton" name="button_cap_square">
           <property name="toolTip">
            <string>Square</string>
           </property>
           <property name="text">
            <string>Square</string>
           </property>
           <property name="icon">
            <iconset theme="stroke-cap-square">
             <normaloff>.</normaloff>.</iconset>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Join</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <layout class="QHBoxLayout" name="layout_join">
         <item>
          <widget class="QToolButton" name="button_join_bevel">
           <property name="toolTip">
            <string>Bevel</string>
           </property>
           <property name="text">
            <string>Bevel</string>
           </property>
           <property name="icon">
            <iconset theme="stroke-join-bevel">
             <normaloff>.</normaloff>.</iconset>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QToolButton" name="button_join_round">
           <property name="toolTip">
            <string>Round</string>
           </property>
           <property name="text">
            <string>Round</string>
           </property>
           <property name="icon">
            <iconset theme="stroke-join-round">
             <normaloff>.</normaloff>.</iconset>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QToolButton" name="button_join_miter">
           <property name="toolTip">
            <string>Miter</string>
           </property>
           <property name="text">
            <string>Miter</string>
           </property>
           <property name="icon">
            <iconset theme="stroke-join-miter">
             <normaloff>.</normaloff>.</iconset>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="spin_miter">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Miter Limit</string>
           </property>
           <property name="decimals">
            <number>2</number>
           </property>
           <property name="maximum">
            <double>999.990000000000009</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Color</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QScrollArea" name="scrollArea">
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>240</width>
            <height>144</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="glaxnimate::gui::ColorSelector" name="color_selector" native="true"/>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <attribute name="title">
       <string>Dashes</string>
      </attribute>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>64</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>128</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>glaxnimate::gui::ColorSelector</class>
   <extends>QWidget</extends>
   <header>widgets/shape_style/color_selector.hpp</header>
   <container>1</container>
   <slots>
    <signal>secondary_color_changed(QColor)</signal>
    <signal>current_color_changed(QColor)</signal>
    <signal>current_color_committed(QColor)</signal>
    <signal>current_color_cleared()</signal>
    <slot>set_secondary_color(QColor)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>button_join_miter</sender>
   <signal>toggled(bool)</signal>
   <receiver>spin_miter</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>244</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_cap_butt</sender>
   <signal>clicked()</signal>
   <receiver>StrokeStyleWidget</receiver>
   <slot>check_cap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>113</y>
    </hint>
    <hint type="destinationlabel">
     <x>91</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_cap_round</sender>
   <signal>clicked()</signal>
   <receiver>StrokeStyleWidget</receiver>
   <slot>check_cap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>167</x>
     <y>113</y>
    </hint>
    <hint type="destinationlabel">
     <x>116</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_cap_square</sender>
   <signal>clicked()</signal>
   <receiver>StrokeStyleWidget</receiver>
   <slot>check_cap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>113</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>91</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_join_miter</sender>
   <signal>clicked()</signal>
   <receiver>StrokeStyleWidget</receiver>
   <slot>check_join()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>243</x>
     <y>141</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_join_bevel</sender>
   <signal>clicked()</signal>
   <receiver>StrokeStyleWidget</receiver>
   <slot>check_join()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>128</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_join_round</sender>
   <signal>clicked()</signal>
   <receiver>StrokeStyleWidget</receiver>
   <slot>check_join()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>167</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>261</x>
     <y>180</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_stroke_width</sender>
   <signal>valueChanged(double)</signal>
   <receiver>StrokeStyleWidget</receiver>
   <slot>check_width(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>140</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>65</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_miter</sender>
   <signal>valueChanged(double)</signal>
   <receiver>StrokeStyleWidget</receiver>
   <slot>check_miter(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>193</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>168</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>color_selector</sender>
   <signal>current_color_changed(QColor)</signal>
   <receiver>StrokeStyleWidget</receiver>
   <slot>check_color(QColor)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>261</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>color_selector</sender>
   <signal>current_color_committed(QColor)</signal>
   <receiver>StrokeStyleWidget</receiver>
   <slot>color_committed(QColor)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>111</x>
     <y>111</y>
    </hint>
    <hint type="destinationlabel">
     <x>261</x>
     <y>121</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_stroke_width</sender>
   <signal>editingFinished()</signal>
   <receiver>StrokeStyleWidget</receiver>
   <slot>commit_width()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>140</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>261</x>
     <y>35</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>color_selector</sender>
   <signal>current_color_cleared()</signal>
   <receiver>StrokeStyleWidget</receiver>
   <slot>clear_color()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>171</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>447</x>
     <y>168</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>check_join()</slot>
  <slot>check_cap()</slot>
  <slot>check_color(QColor)</slot>
  <slot>check_width(double)</slot>
  <slot>check_miter(double)</slot>
  <slot>color_committed(QColor)</slot>
  <slot>commit_width()</slot>
  <slot>clear_color()</slot>
 </slots>
</ui>
